flink-ml-lib/src/test/java/org/apache/flink/ml/feature/RandomSplitterTest.java

```diff
@@ -0,0 +1,145 @@
+
+        env = StreamExecutionEnvironment.getExecutionEnvironment(config);
+        env.setParallelism(1);
+        env.enableCheckpointing(100);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+
+        tEnv = StreamTableEnvironment.create(env);
+    }
+
+    private Table getTable(int size) {
+        DataStreamSource<Long> dataStream = env.fromSequence(0L, size);
+        return tEnv.fromDataStream(dataStream);
+    }
+
+    @Test
+    public void testParam() {
+        RandomSplitter splitter = new RandomSplitter();
+        splitter.setWeights(0.3, 0.4);
+        assertArrayEquals(new Double[] {0.3, 0.4}, splitter.getWeights());
+    }
+
+    @Test
+    public void testOutputSchema() {
+        Table tempTable =
+                tEnv.fromDataStream(env.fromElements(Row.of("", "")))
+                        .as("test_input", "dummy_input");
+
+        RandomSplitter splitter = new RandomSplitter().setWeights(0.5, 0.1);
+        Table[] output = splitter.transform(tempTable);
+        assertEquals(2, output.length);
+        for (Table table : output) {
+            assertEquals(
+                    Arrays.asList("test_input", "dummy_input"),
+                    table.getResolvedSchema().getColumnNames());
+        }
+    }
+
+    @Test
+    public void testWeights() throws Exception {
+        Table data = getTable(1000);
+        RandomSplitter splitter = new RandomSplitter().setWeights(2.0, 1.0, 2.0);
+        Table[] output = splitter.transform(data);
+
+        List<Row> result0 = IteratorUtils.toList(tEnv.toDataStream(output[0]).executeAndCollect());
+        List<Row> result1 = IteratorUtils.toList(tEnv.toDataStream(output[1]).executeAndCollect());
+        List<Row> result2 = IteratorUtils.toList(tEnv.toDataStream(output[2]).executeAndCollect());
+        assertEquals(result0.size() / 400.0, 1.0, 0.1);
+        assertEquals(result1.size() / 200.0, 1.0, 0.1);
+        assertEquals(result2.size() / 400.0, 1.0, 0.1);
+        verifyResultTables(data, output);
+    }
+
+    @Test
+    public void testSaveLoadAndTransform() throws Exception {
+        Table data = getTable(2000);
+        RandomSplitter randomSplitter = new RandomSplitter().setWeights(4.0, 6.0);
+
+        RandomSplitter splitterLoad =
+                TestUtils.saveAndReload(
+                        tEnv, randomSplitter, TEMPORARY_FOLDER.newFolder().getAbsolutePath());
+
+        Table[] output = splitterLoad.transform(data);
+        List<Row> result0 = IteratorUtils.toList(tEnv.toDataStream(output[0]).executeAndCollect());
+        List<Row> result1 = IteratorUtils.toList(tEnv.toDataStream(output[1]).executeAndCollect());
+        assertEquals(result0.size() / 800.0, 1.0, 0.1);
+        assertEquals(result1.size() / 1200.0, 1.0, 0.1);
+        verifyResultTables(data, output);
+    }
+
+    private void verifyResultTables(Table input, Table[] output) throws Exception {
```

Comment:
The logic seems a bit confusing here. How about we directly compare the elements in `input` and `output` by adding them to two lists?